packages/type-utils/src/typeOrValueSpecifiers/typeDeclaredInPackage.ts

```diff
@@ -0,0 +1,55 @@
+import * as ts from 'typescript';
+
+function findParentModuleDeclaration(
+  node: ts.Node,
+): ts.ModuleDeclaration | undefined {
+  switch (node.kind) {
+    case ts.SyntaxKind.ModuleDeclaration:
+      return node as ts.ModuleDeclaration;
+    case ts.SyntaxKind.SourceFile:
+      return undefined;
+    default:
+      return findParentModuleDeclaration(node.parent);
+  }
+}
+
+function typeDeclaredInDeclareModule(
```

Comment:
This is the main added logic in this PR.

cc @marekdedic @RebeccaStevens as previous authors around the TypeOrValueSpecifier tooling. Sorry it took me so long to follow up on this point!